policies/templates/gcp_bq_dataset_location_v1.yaml

```diff
@@ -0,0 +1,182 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This template is for policies restricting the locations 
+# of BigQuery datasets to specific locations in GCP. It supports 
+# allowlist or denylist modes, as well as exempting selected 
+# datasets from the list.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-bq-dataset-location-v1
+  annoations:
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPBigQueryDatasetLocationConstraintV1
+        plural: gcpbigquerydatasetlocationconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            mode:
+              type: string
+              enum: [denylist, allowlist]
+              description: "String identifying the operational mode, allowlist or denylist. In allowlist mode, 
+              datasets are only allowed in the locations specified in the 'locations' parameter. In denylist mode, 
+              resources are allowed in all locations except those listed in the 'locations' parameter."
+            exemptions:
+              type: array
+              items: string
+              description: "Array of datasets to exempt from location restriction. String values in the array should 
+              correspond to the full name values of exempted datasets."
+            locations:
+              type: array
+              items: string
+              description: "Array of location names to be allowed or denied. Should be the simple form of the location name, 
```

Comment:
nit: instead of "simple form", can we be more specific and mention this is either the regional location (`us-west2`) or multi-regional location (`EU`)? Also including a reference to https://cloud.google.com/bigquery/docs/locations would be useful.

Reply:
Fixed.